packages/flutter/lib/src/material/drawer.dart

```diff
@@ -501,8 +501,9 @@ class DrawerControllerState extends State<DrawerController> with SingleTickerPro
     if (widget.scrimColor != oldWidget.scrimColor) {
       _scrimColorTween = _buildScrimColorTween();
     }
-    if (widget.isDrawerOpen != oldWidget.isDrawerOpen && !_controller.isAnimating) {
-      _controller.value = widget.isDrawerOpen ? 1.0 : 0.0;
+    final bool drawerOpen = widget.isDrawerOpen;
```

Comment:
I usually like introducing variables to make code more understandable and readable. But in that case `widget.idDrawerOpen` is already readable and `widget.isDrawerOpen != oldWidget.isDrawerOpen` seems more understandable than `drawerOpen != oldWidget.isDrawerOpen`

Reply:
For the first one, it doesn't have to be `late` does it?

I vote for 1 or 2, but really I think any way works.

Reply:
Just had another idea that I really like!

```dart
if (_controller.status.isAnimating) {
  return; // Don't snap the drawer open or shut while the user is dragging.
}
if (widget.isDrawerOpen != oldWidget.isDrawerOpen) {
  _controller.value = widget.isDrawerOpen ? 1.0 : 0.0;
}

```